src/connection_string.ts

```diff
@@ -1572,5 +1031,33 @@ export const OPTIONS = {
   zlibCompressionLevel: {
     default: 0,
     type: 'int'
-  }
+  },
+  // Custom types for modifying core behavior
+  connectionType: { type: 'any' },
+  srvPoller: { type: 'any' },
+  /// Accepted NodeJS Options
```

Comment:
There's a trade off here to make it more automatic, I can use the list of options we support in `LEGAL_TLS_SOCKET_OPTIONS` for example and loop over it doing something like `OPTIONS[key] = {type: 'any'}` and that would get us these options defined in this registry, but then we would lack the type assertion that is currently in place. The type of the OPTIONS registry is derived from MongoClientOptions so any options we support there must be defined here in the registry.

Either way (even before this work) we would need to make an explicit change to accept a new or changed option since we have the permitted options filter that gets applied in our connect logic. I hope by relying on the definitions in `@types/node` we'll actually be able to just add the new option to our permitted list, and then an unfortunate but coerced by the compiler `newOption: { type: 'any' }` line to this struct.

Reply:
Thanks for the explanation, makes sense now.